docassemble/AssemblyLine/al_general.py

```diff
@@ -296,6 +366,22 @@ def on_one_line(
             output += ", " + country_name(self._get_country())
         return output
 
+    def normalized_address(self) -> Union[Address, "ALAddress"]:
+        """
+        Try geocoding the address, and if it succeeds, return the "long" normalized version of
+        the address. All methods are still available, such as my_address.normalized_address().block(), etc.,
+        but note that this will be a standard Address object, not an ALAddress object.
+
+        If geocoding fails, return the version of the address as entered by the user instead.
+        """
+        try:
+            self.geocode()
```

Comment:
Will `geocode()` actually do the geo coding each time it's called? Or just once and then uses a cached version

Reply:
Just once:

>Normally, calling geocode() on an object (without an address parameter) after geocoding has been performed has no effect. This allows you to safely insert a call to geocode() in your interview logic without running up a large bill for your [Google API](https://developers.google.com/api-client-library/python/apis/) usage.

https://docassemble.org/docs/objects.html#Address